### PR TITLE
feat: Updated logger version

### PR DESCRIPTION
**Pull request by Evgenia-bit**
## Changes
- Updated logger version

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [x] Required labels marked
- [ ] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [ ] Attached videos/screenshots demonstrating the fix/feature.
- [x] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: lib/features/app/di/app_scope.dart [hunk 1]
```diff
@@ -116,6 +127,28 @@ class AppScope implements IAppScope {
...
+  /// Add strategy to logger for save logs history for qa environment.
+  Future<surf.LogStrategy> _logHistoryStrategy() async {
+    final file = await const LogHistoryServiceImpl().logHistoryFile();
+    final logger = Logger(
+      output: FileCustomOutput(file: file),
+      printer: PrettyPrinter(lineLength: 80, noBoxingByDefault: true),
+    );
+
+    return LogHistoryStrategy(logger);
+  }
```
Comment (maxim-sysoev) [inline lib/features/app/di/app_scope.dart:151]: It's better to do this in another file

File: lib/util/log_strategy/crashlytics_log_strategy.dart [hunk 2]
```diff
@@ -3,33 +3,20 @@ import 'package:flutter/foundation.dart';
 import 'package:surf_logger/surf_logger.dart';
 
 /// Strategy for sending logs to Crashlytics.
-class CrashlyticsRemoteLogStrategy extends RemoteUserLogStrategy {
+class CrashlyticsLogStrategy extends LogStrategy {
   FirebaseCrashlytics get _crashlytics => FirebaseCrashlytics.instance;
 
-  @override
+  /// Add user info.
```
Comment (maxim-sysoev) [inline lib/util/log_strategy/crashlytics_log_strategy.dart:9]: It is worth adding both username and email to crashlytics parameters

File: lib/features/app/di/app_scope.dart [hunk 3]
```diff
@@ -17,7 +17,10 @@ import 'package:flutter_template/features/navigation/service/router.dart';
 import 'package:flutter_template/persistence/storage/theme_storage/theme_storage.dart';
 import 'package:flutter_template/persistence/storage/theme_storage/theme_storage_impl.dart';
 import 'package:flutter_template/util/default_error_handler.dart';
+import 'package:flutter_template/util/log_strategy/debug_log_strategy.dart';
+import 'package:flutter_template/util/log_strategy/log_history_strategy.dart';
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:surf_logger/surf_logger.dart' as surf;
```
Comment (dzmitry-struk-surf) [inline lib/features/app/di/app_scope.dart:24]: `surf` is not a defining name for the logger, remove the `as`, please.

File: lib/features/app/di/app_scope.dart [hunk 4]
```diff
@@ -116,6 +125,15 @@ class AppScope implements IAppScope {
   Future<void> _onThemeModeChanged() async {
     await _themeModeStorage.saveThemeMode(mode: _themeService.currentThemeMode);
   }
+
+  Future<void> _initLogger() async {
+    _logger = surf.Logger.withStrategies({
+      // TODO(init-project): Initialize CrashlyticsRemoteLßogStrategy.
```
Comment (dzmitry-struk-surf) [inline lib/features/app/di/app_scope.dart:131]: CrashlyticsRemoteL**ß**ogStrategy remove that strange symbol, please.

**Review by dzmitry-struk-surf — CHANGES_REQUESTED** [review]

**Review by dzmitry-struk-surf — APPROVED** [review]

**Pull request merged**